pkg/bpf/exporter.go

```diff
@@ -98,16 +86,11 @@ type perfCounter struct {
 	EvConfig int
 }
 
-func (e *exporter) GetEnabledBPFHWCounters() []string {
-	return e.enabledHardwareCounters
-}
-
-func (e *exporter) GetEnabledBPFSWCounters() []string {
-	return e.enabledSoftwareCounters
-}
-
-func (e *exporter) HardwareCountersEnabled() bool {
-	return e.hardwareCountersEnabled
+func (e *exporter) SupportedMetrics() SupportedMetrics {
+	return SupportedMetrics{
+		HardwareCounters: e.enabledHardwareCounters,
+		SoftwareCounters: e.enabledSoftwareCounters,
```

Comment:
Could we return a copy instead of pointer to the real object to avoid and unintended side effects.

Reply:
Good catch - I've been doing Rust too long 🦀 and had forgotten that Go maps are passed by reference 😮‍💨